File: src/main/java/com/back/domain/member/member/repository/MemberRepository.java
package com.back.domain.member.member.repository;

import com.back.domain.member.member.entity.Member;
import com.back.domain.member.member.enums.MemberStatus;
import org.springframework.data.jpa.repository.JpaRepository;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Optional;
import java.util.Set;

public interface MemberRepository extends JpaRepository<Member, Integer> {
    Optional<Member> findByUsername(String username);

    Optional<Member> findByApiKey(String apiKey);

    Optional<Member> findByNickname(String nickname);

    List<Member> findByApiKeyIn(Set<String> apiKeys);
    List<Member> findByStatusAndSuspendAtBefore(
            MemberStatus status,
            LocalDateTime time
    );

}

File: src/main/java/com/back/domain/member/member/repository/MemberRepository.kt
package com.back.domain.member.member.repository

import com.back.domain.member.member.entity.Member
import com.back.domain.member.member.enums.MemberStatus
import org.springframework.data.jpa.repository.JpaRepository
import java.time.LocalDateTime
import java.util.*

interface MemberRepository : JpaRepository<Member, Int> {
    fun findByUsername(username: String): Optional<Member>

    fun findByApiKey(apiKey: String): Optional<Member>

    fun findByNickname(nickname: String): Optional<Member>

    fun findByApiKeyIn(apiKeys: MutableSet<String>): MutableList<Member>
    fun findByStatusAndSuspendAtBefore(
        status: MemberStatus,
        time: LocalDateTime
    ): MutableList<Member>
}

